src/java/org/apache/cassandra/index/sai/disk/vector/CassandraOnHeapGraph.java

```diff
@@ -80,14 +85,16 @@ public class CassandraOnHeapGraph<T>
     private final VectorSimilarityFunction similarityFunction;
     private final ConcurrentMap<float[], VectorPostings<T>> postingsMap;
     private final NonBlockingHashMapLong<VectorPostings<T>> postingsByOrdinal;
+    private final NonBlockingHashMap<T, float[]> vectorCache;
+    private final boolean cacheVectors;
     private final AtomicInteger nextOrdinal = new AtomicInteger();
     private volatile boolean hasDeletions;
 
     /**
      * @param termComparator the vector type -- passed as AbstractType for caller's convenience
      * @param indexWriterConfig
      */
-    public CassandraOnHeapGraph(AbstractType<?> termComparator, IndexWriterConfig indexWriterConfig)
+    public CassandraOnHeapGraph(AbstractType<?> termComparator, IndexWriterConfig indexWriterConfig, boolean shouldCacheVectors)
```

Comment:
i'd prefer a more high-level name for the new bool.  `forMemtable`, `forSearching`, or the opposite `writeOnly` (or maybe none of these #collaborating)